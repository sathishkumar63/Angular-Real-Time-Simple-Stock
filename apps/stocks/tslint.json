{
  "extends": "../../tslint.json",
  "rules": {
    "directive-selector": [true, "attribute", "stocks", "camelCase"],
    "component-selector": [true, "element", "stocks", "kebab-case"]
  },
  "linterOptions": {
    "exclude": ["!**/*"]
  }
}
