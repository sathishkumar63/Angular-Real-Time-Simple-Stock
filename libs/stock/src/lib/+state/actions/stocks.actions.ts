import { Action } from '@ngrx/store';
import { IStock } from '../../models';

export enum StocksActionTypes {
  GetStocks = '[Stocks] Get',
  GetStocksSuccess = '[Stocks] Get Success',
  GetStocksFail = '[Stocks] Get Fail',
  SetStocks = '[Stocks] Set',
}

export class GetStocks implements Action {
  readonly type = StocksActionTypes.GetStocks;
}

export class GetStocksSuccess implements Action {
  readonly type = StocksActionTypes.GetStocksSuccess;
  constructor(public payload: IStock[]) {}
}

export class GetStocksFail implements Action {
  readonly type = StocksActionTypes.GetStocksFail;
  constructor(public payload: Error) {}
}

export class SetStocks implements Action {
  readonly type = StocksActionTypes.SetStocks;
  constructor(public payload: IStock) {}
}

export type StocksAction =
  | GetStocks
  | GetStocksSuccess
  | GetStocksFail
  | SetStocks;
