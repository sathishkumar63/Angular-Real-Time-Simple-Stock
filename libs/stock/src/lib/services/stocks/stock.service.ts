import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { IStock, QUOTE_URL } from '../../models';

@Injectable({
  providedIn: 'root',
})
export class StockService {
  constructor(private http: HttpClient) {}

  getStocks(symbols: string[]): Observable<any> {
    // HttpRequest has been blocked by CORS policy. So, add cors-anywhere.herokuapp.com to this link.
    const uri = `https://cors-anywhere.herokuapp.com/${QUOTE_URL}symbols=${symbols?.join()}`;
    return this.http.get(uri);
  }

  formatStocksResponse(stocks: any[]): IStock[] {
    return stocks?.map((stock) => ({
      name: stock.displayName,
      symbol: stock.symbol,
      currency: stock.currency,
      currentPrice: stock.regularMarketPrice,
      dayHighPrice: stock.regularMarketDayHigh,
      dayLowPrice: stock.regularMarketDayLow,
      fiftyTwoWeekLow: stock.fiftyTwoWeekLow,
      fiftyTwoWeekHigh: stock.fiftyTwoWeekHigh,
      regularMarketVolume: stock.regularMarketVolume,
    }));
  }
}
