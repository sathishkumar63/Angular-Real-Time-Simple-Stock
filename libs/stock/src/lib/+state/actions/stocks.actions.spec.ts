import * as fromStocks from './stocks.actions';

describe('GetStocks', () => {
  it('should create an instance', () => {
    const action = new fromStocks.GetStocks();
    expect(action).toEqual({
      type: fromStocks.StocksActionTypes.GetStocks,
    });
    expect(action).toMatchSnapshot();
  });
});

describe('GetStocksSuccess', () => {
  it('should create an instance', () => {
    const action = new fromStocks.GetStocksSuccess([]);
    expect(action).toEqual({
      type: fromStocks.StocksActionTypes.GetStocksSuccess,
      payload: [],
    });
    expect(action).toMatchSnapshot();
  });
});

describe('GetStocksFail', () => {
  it('should create an instance', () => {
    const action = new fromStocks.GetStocksFail();
    expect(action).toBeTruthy();
    expect(action).toMatchSnapshot();
  });
});
