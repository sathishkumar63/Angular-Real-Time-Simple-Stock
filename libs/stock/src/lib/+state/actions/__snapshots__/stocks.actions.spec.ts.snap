// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetStocks should create an instance 1`] = `
GetStocks {
  "type": "[Stocks] Get",
}
`;

exports[`GetStocksFail should create an instance 1`] = `
GetStocksFail {
  "type": "[Stocks] Get Fail",
}
`;

exports[`GetStocksSuccess should create an instance 1`] = `
GetStocksSuccess {
  "payload": Array [],
  "type": "[Stocks] Get Success",
}
`;
